pkg/deploy/api/helpers.go

```diff
@@ -68,40 +83,61 @@ func TemplateImageForContainer(pod *kapi.PodSpec, triggerFn TriggeredByFunc, con
 		if container.Name != containerName {
 			continue
 		}
-		var ref imageapi.DockerImageReference
-		if trigger, ok := triggerFn(container); ok {
-			trigger.Image = container.Image
-			trigger.Container = container
-			return trigger, nil
-		}
-		ref, err := imageapi.ParseDockerImageReference(container.Image)
-		if err != nil {
-			return TemplateImage{Image: container.Image, Container: container}, err
+		return templateImageForContainer(container, triggerFn)
+	}
+	for i := range pod.InitContainers {
+		container := &pod.InitContainers[i]
+		if container.Name != containerName {
+			continue
 		}
-		return TemplateImage{Image: container.Image, Ref: &ref, Container: container}, nil
+		return templateImageForContainer(container, triggerFn)
 	}
 	return TemplateImage{}, fmt.Errorf("no container %q found", containerName)
 }
 
+// eachTemplateImage invokes triggerFn and fn on the provided container.
+func eachTemplateImage(container *kapi.Container, triggerFn TriggeredByFunc, fn func(TemplateImage, error)) {
+	var ref imageapi.DockerImageReference
```

Comment:
This can be:
```go
func eachTemplateImage(container *kapi.Container, triggerFn TriggeredByFunc, fn func(TemplateImage, error)) {
    image, err := templateImageForContainer(container, triggerFn)
    fn(image, err)
}
```